From 90344c323e0c251cddd52cdcd9e705763bb10aa8 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Sat, 13 Jul 2024 18:06:42 +0300
Subject: [PATCH] tests: Set sta_wds in multi_ap_backhaul_shared_bss

This is needed to add the WDS interface for a STA with stricter
implementation checks.

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 tests/hwsim/test_multi_ap.py | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/tests/hwsim/test_multi_ap.py b/tests/hwsim/test_multi_ap.py
index 8bfcbda21..e1dfa568c 100644
--- a/tests/hwsim/test_multi_ap.py
+++ b/tests/hwsim/test_multi_ap.py
@@ -32,15 +32,19 @@ def test_multi_ap_association_shared_bss(dev, apdev):
 
 def test_multi_ap_backhaul_shared_bss(dev, apdev):
     """Multi-AP backhaul to backhaul+fronthaul BSS"""
-    hapd = run_multi_ap_association(dev, apdev, 3, wait_connect=False)
+    hapd = run_multi_ap_association(dev, apdev, 3, wait_connect=False,
+                                    wds_sta=True)
     ev = hapd.wait_event(["WDS-STA-INTERFACE-ADDED"], timeout=10)
     if ev is None:
         raise Exception("No WDS-STA-INTERFACE-ADDED event seen")
 
-def run_multi_ap_association(dev, apdev, multi_ap, wait_connect=True):
+def run_multi_ap_association(dev, apdev, multi_ap, wait_connect=True,
+                             wds_sta=False):
     params = hostapd.wpa2_params(ssid="multi-ap", passphrase="12345678")
     if multi_ap:
         params["multi_ap"] = str(multi_ap)
+    if wds_sta:
+        params["wds_sta"] = "1"
     hapd = hostapd.add_ap(apdev[0], params)
 
     dev[0].connect("multi-ap", psk="12345678", scan_freq="2412",
-- 
2.43.0

